test/acceptance/tests/basic_test.go

```diff
@@ -171,11 +166,24 @@ func TestBasic(t *testing.T) {
 				tags["serverless.consul.hashicorp.com/v1alpha1/lambda/namespace"] = namespace
 			}
 
-			lambdaTerraformOptions.Vars = map[string]interface{}{
-				"tags":   tags,
-				"name":   lambdaServiceName,
-				"region": config.Region,
-			}
+			lambdaTerraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+				TerraformDir: "./lambda",
+				NoColor:      true,
+				Vars: map[string]interface{}{
+					"tags":   tags,
+					"name":   lambdaServiceName,
+					"region": config.Region,
+				},
+			})
+
+			t.Cleanup(func() {
```

Comment:
I moved this down here because if the testing above fails for some reason we don't want to attempt to `terraform destroy` the lambda workspace because it was never applied. There ended up being misleading error messages in the logs about failing to destroy the lambda resources when the test failed prior to creating them.